### PR TITLE
Use new io-lib.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/web_socket.toit [hunk 1]
```diff
@@ -128,18 +121,14 @@ class WebSocket:
...
-    len := pending_[1] & 0x7f
+
+    reader := socket_.in
+    if not reader.try-ensure-buffered 1: return null
+
+    control_bits := reader.read-byte
+    masking_length_byte := reader.read-byte
+    masking := (masking_length_byte & MASKING_FLAG_) != 0
+    len := masking_length_byte & 0x7f
     header_size_needed := ?
```
Comment (erikcorry) [inline src/web_socket.toit:132]: header_size_needed is no longer needed?

**Review by erikcorry — APPROVED** [review]

File: src/web_socket.toit (hunk 1 above)
Comment (floitsch) [reply src/web_socket.toit]: done.

File: src/chunked.toit [hunk 2]
```diff
@@ -55,34 +60,38 @@ class ChunkedReader implements reader.Reader:
...
+class ChunkedWriter extends Object with io.CloseableWriter:
   static CRLF_ ::= "\r\n"
 
   connection_/Connection? := null
-  writer_/writer.Writer
+  writer_/io.Writer
 
   // We don't know the amount of data ahead of time, so it may already be done.
-  is_done -> bool:
+  is_done_ -> bool:
```
Comment (kasperl) [inline src/chunked.toit:78]: It is a little bit funky that this comes before the constructor.

File: src/connection.toit [hunk 3]
```diff
@@ -47,14 +42,15 @@ class Connection:
   close -> none:
     if socket_:
       socket_.close
+      // TODO(florian): should we close the writer?
```
Comment (kasperl) [inline src/connection.toit:45]: That feels like the wrong todo. Maybe rephrase to something like "figure out if we actually want to close the writer here" or "should we avoid closing the writer".

File: src/connection.toit [hunk 4]
```diff
@@ -143,68 +150,72 @@ class Connection:
...
       return null
-    index_of_first_space := reader_.index_of_or_throw ' '
-    method := reader_.read_string (index_of_first_space)
-    reader_.skip 1
-    path := reader_.read_string (reader_.index_of_or_throw ' ')
-    reader_.skip 1
-    version := reader_.read_string (reader_.index_of_or_throw '\r')
-    reader_.skip 1
-    if reader_.read_byte != '\n': throw "FORMAT_ERROR"
+    index_of_first_space := reader.index_of ' ' --throw-if-missing
```
Comment (kasperl) [inline src/connection.toit:158]: Looking at this, I think I would have preferred `--throw-if-absent`.

File: src/connection.toit [hunk 5]
```diff
@@ -222,111 +233,125 @@ class Connection:
...
+  size/int
 
-  size -> int:
-    return content_length
+  constructor .connection_ .reader_ .size:
 
-  read -> ByteArray?:
-    if remaining_length_ <= 0:
+  /**
+  Deprecated. Use $size instead.
```
Comment (kasperl) [inline src/connection.toit:291]: Maybe implement `content-size` and recommend using that instead?

File: src/response.toit [hunk 6]
```diff
@@ -23,10 +22,16 @@ class Response:
...
-  body/reader.Reader
+  body/io.Reader
 
   constructor .connection_ .version .status_code .status_message .headers .body:
 
+  /**
+  The length of the response body, if known.
+  */
+  content_length -> int?:
+    return body.size
```
Comment (kasperl) [inline src/response.toit:33]: `content-size`, perhaps?
